lib/kv-redis.js

```diff
@@ -357,14 +370,28 @@ function(modelName, filter, options, callback) {
   function fetchFromRedis(cb) {
     if (cursor === null) cursor = '0';
 
-    ModelKeyComposer.compose(modelName, keyPattern, function(err, pattern) {
-      if (err) return cb(err);
-      self.execute('SCAN', [cursor, 'MATCH', pattern], function(err, result) {
+    ModelKeyComposer.compose(customModelName, keyPattern,
+      function(err, pattern) {
         if (err) return cb(err);
-        cursor = result[0].toString('utf8');
-        cache = result[1].map(function(it) { return it.toString('utf8'); });
-        getNextKey(cb);
+        self.execute('SCAN', [cursor, 'MATCH', pattern], function(err, result) {
+          if (err) return cb(err);
+          cursor = result[0].toString('utf8');
+          cache = result[1].map(function(it) { return it.toString('utf8'); });
+          getNextKey(cb);
+        });
       });
-    });
   }
 };
+
+/**
+ * Get model name for a given model
+ * @param {String} modelName The model name
+ * @returns {String} model name
+ */
+RedisKeyValueConnector.prototype.modelName = function(modelName) {
+  const modelClass = this._models[modelName];
+  if (modelClass && modelClass.settings && modelClass.settings.redis) {
```

Comment:
The connector name is `kv-redis`, therefore I think the settings key should be called `kv-redis` too, for consistency with other connectors. Here is the method in the base `Connector` class that you should use to obtain connector-specific settings:

https://github.com/strongloop/loopback-connector/blob/deb461ea62e6226f053774fb10c1f023d6061f8c/lib/connector.js#L126-L129

See also how is this method used by the SqlConnector to obtain table name:

https://github.com/strongloop/loopback-connector/blob/deb461ea62e6226f053774fb10c1f023d6061f8c/lib/sql.js#L315-L322

Reply:
There is test that fail if use `getConnectorSpecificSettings` method because access to method connector without create a model class. See the [test](https://github.com/strongloop/loopback-connector-kv-redis/blob/8a9700674a31f95d76ce31183464b96f647a3e62/test/integration/packer.integration.js#L22) for more detail

Reply:
AFAICT, the test is not relying on the model not being defined. I assume the author omitted model definition because it was not required to make the test pass at that time.

I checked few of the "regular" test files in juggler's [test/kvao](https://github.com/strongloop/loopback-datasource-juggler/tree/master/test/kvao), they are all creating a new model class before calling any connector methods.

Let's fix the problematic test and ensure it defines a new model before calling any KVAO methods on the datasource or connector.

The test model can be empty (with no properties) as long as it has `strict` mode disabled in model settings.